### PR TITLE
Exit codes implementation

### PR DESCRIPTION
**Pull request by 0xInfection**
This PR addresses:
- [x] Exit codes across `svmap`, `svwar`, `svcrack` as discussed.
- [x] Some code refactoring and fixing other unbound errors.

__Example:__
Few examples for `svwar`:
- network connectivity errors
  ```
  $ sipvicious_svwar demo
  ERROR:TakeASip:socket error: [Errno -3] Temporary failure in name resolution
  WARNING:root:found nothing
  
  $ echo $?
  30
  ```
- syntax errors
  ```
  $ sipvicious_svwar udp://demo.sipvicious.pro:
  ERROR:TakeASip:port should strictly be an integer between 1 and 65535
  
  $ echo $?
  10
  ```
- security issue
  ```
  $ sipvicious_svwar udp://demo.sipvicious.pro:5060 -e 1000
  +-----------+----------------+
  | Extension | Authentication |
  +===========+================+
  | 1000      | reqauth        |
  +-----------+----------------+
  
  $ echo $?
  40
  ```

### CODE REVIEW

**Review by sandrogauci — COMMENTED** [review]
```
sipvicious_svmap demo.sipvicious.pro; echo $?
+---------------------+---------------------------------+
| SIP Device          | User Agent                      |
+=====================+=================================+
| 172.104.142.43:5060 | kamailio (5.4.4 (x86_64/linux)) |
+---------------------+---------------------------------+
30
```
This 30 seems wrong.

**Comment by sandrogauci** [comment]
> ```
> sipvicious_svmap demo.sipvicious.pro; echo $?
> +---------------------+---------------------------------+
> | SIP Device          | User Agent                      |
> +=====================+=================================+
> | 172.104.142.43:5060 | kamailio (5.4.4 (x86_64/linux)) |
> +---------------------+---------------------------------+
> 30
> ```
> 
> This 30 seems wrong.

Once this is sorted, a decent amount of automated tests should find the obvious issues.

File: sipvicious/svcrack.py [hunk 1]
```diff
@@ -546,8 +577,11 @@ def main():
         sipvicious.start()
         if exportpath is not None:
             open(os.path.join(exportpath, 'closed'), 'w').close()
+
     except KeyboardInterrupt:
         logging.warning('caught your control^c - quiting')
+        __exitcode__ = resolveexitcode(30, __exitcode__)
```
Comment (sandrogauci) [inline sipvicious/svcrack.py:583]: keyboard interrupt is not a network error.

No exit code should be set on keyboard interrupt; any previous exit codes set should follow through to the final `exit()` whether or not this exception is triggered.

File: sipvicious/svmap.py [hunk 2]
```diff
@@ -423,92 +450,101 @@ def main():
...
-        except AssertionError as err:
-            logging.critical(err)
-            exit(1)
+        sipvicious.start()
         if exportpath is not None:
             open(os.path.join(exportpath,'closed'),'w').close()
+
     except KeyboardInterrupt:
         logging.warning( 'caught your control^c - quiting' )
+        __exitcode__ = resolveexitcode(20, __exitcode__)
```
Comment (sandrogauci) [inline sipvicious/svmap.py:535]: same here - don't set exit code because of keyboard interrupt

File: sipvicious/svcrack.py (hunk 1 above)
Comment (0xInfection) [reply sipvicious/svcrack.py]: taken care in 3d909a7.

File: sipvicious/svmap.py (hunk 2 above)
Comment (0xInfection) [reply sipvicious/svmap.py]: taken care in 3d909a7.

**Review by sandrogauci — APPROVED** [review]
great job!

**Pull request merged**